DependencyInjection/HttplugExtension.php

```diff
@@ -207,4 +217,20 @@ private function configureAuthentication(ContainerBuilder $container, array $con
                 ->addArgument(new Reference($authServiceKey));
         }
     }
+
+    /**
+     * Verifies that Puli bundle is installed.
+     *
+     * @param ContainerBuilder $container
+     *
+     * @throws \Exception
+     */
+    private function verifyPuliInstalled(ContainerBuilder $container)
+    {
+        if (false === $container->has('puli.discovery')) {
```

Comment:
In a bundle's extension class you can not see services registered in other bundles. That is why `$container->has('puli.discovery')` will always be false. 

I suggest to save the $useDiscovery variable in a container parameter and then handle the rest in a compiler pass. 

If we check if discovery is necessary in a compiler pass we need to inspect `httplug.client.default`, `httplug.message_factory.default` etc to see if they are using the standard class or not. That may be more elegant because we do not use container parameters but I believe it requires more hard coded values.